app/src/main/java/org/simple/clinic/patient/Patient.kt

```diff
@@ -366,6 +366,30 @@ data class Patient(
       WHERE uuid = :patientUuid
     """)
     abstract fun contactPatientProfileImmediate(patientUuid: UUID): ContactPatientProfile
+
+    @Query("""
+      SELECT * FROM (
+        SELECT DISTINCT colonyOrVillage 
+        FROM PatientAddress 
+        WHERE colonyOrVillage IS NOT NULL 
+        AND uuid IN (
+          SELECT DISTINCT P.addressUuid 
+          FROM Patient P
+          WHERE P.registeredFacilityId = :facilityUuid 
+          OR P.assignedFacilityId = :facilityUuid 
+        ) 
+
+        UNION 
+    
+        SELECT DISTINCT fullName 
+        FROM Patient 
+        WHERE fullName IS NOT NULL 
+        AND registeredFacilityId = :facilityUuid 
+        OR assignedFacilityId = :facilityUuid
+        )
+     ORDER BY colonyOrVillage ASC   
+    """)
+    abstract fun fetchColonyOrVillageAndPatientNames(facilityUuid: UUID): List<String>
```

Comment:
Not so sure about the function name, feel like we are doing too many things in a single function.

Reply:
We're loading village and patient names both, do you think `loadVillageAndPatientNames` would be better? @msasikanth @janhavisinghh

Reply:
Yeah, that works `loadPatientAndVillageNames`